ClientRequirements.md

```diff
@@ -1,5 +1,10 @@
 # Platform Requirements for Fluid Framework Client Packages
```

Comment:
For reviewers: this document used to be the "source of truth" of this information. This PR would move that source of truth into the markdown-magic templates directory, but we still want to retain this information at the root of the repo for discoverability.